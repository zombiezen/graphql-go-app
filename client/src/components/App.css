.App {
  background: #f8cdea;
  border: 1px solid #2c2c2c;
  border-radius: 8px;
  box-shadow: 1px 1px 5px 1px rgba(0,0,0,0.2);
  box-sizing: border-box;
  margin: 3rem auto 1rem;
  max-width: 20rem;
  padding: 0.25rem 1rem;
}
.App-heading {
  margin: 0.25rem 0 0.5rem;
}
