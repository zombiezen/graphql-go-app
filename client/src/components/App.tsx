// Copyright 2019 Ross Light
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// SPDX-License-Identifier: Apache-2.0

import { useQuery } from '@apollo/react-hooks';
import React from 'react';
import { Link, Route, Switch } from 'react-router-dom';

import './App.css';
import QUERY from './AppQuery.graphql';

interface QueryData {
  greeting: string;
}

export const App: React.FC<{}> = _props => {
  const { data } = useQuery<QueryData>(QUERY);
  return (
    <main className="App">
      <Switch>
        <Route path="/" exact>
          <h1 className="App-heading">GraphQL Go App</h1>
          {data ? (
            <p className="App-greeting">{data.greeting} (from GraphQL)</p>
          ) : (
            <p className="App-loading">Loading&hellip;</p>
          )}

          <p>
            Try the <a href="/client/playground.html">GraphQL playground</a>!
          </p>
        </Route>
        <Route path="*">
          <h1 className="App-heading">Not Found</h1>
          <p>
            Try heading back <Link to="/">home</Link>?
          </p>
        </Route>
      </Switch>
    </main>
  );
};
