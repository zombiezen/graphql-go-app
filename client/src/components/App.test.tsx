// Copyright 2019 Ross Light
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// SPDX-License-Identifier: Apache-2.0

import { MockedProvider, MockedResponse } from '@apollo/react-testing';
import { mount } from 'enzyme';
import * as React from 'react';
import { act } from 'react-dom/test-utils';
import waitForExpect from 'wait-for-expect';

import { App, QUERY } from './App';
import { MemoryRouter } from 'react-router';

describe('App', () => {
  const mocks: MockedResponse[] = [
    {
      request: {
        query: QUERY,
      },
      result: {
        data: {
          greeting: 'Hello, World!',
        },
      },
    },
  ];
  const Wrapper: React.FC<{children?: any}> = ({children}) => (
    <MockedProvider mocks={mocks} addTypename={false}>
      <MemoryRouter initialEntries={[ { pathname: '/', key: 'xyzzy' } ]}>
        {children}
      </MemoryRouter>
    </MockedProvider>
  );

  it('renders a loading message', () => {
    const component = mount(<App />, {
      wrappingComponent: Wrapper,
    });
    expect(component).toMatchSnapshot();
  });
  it('renders the greeting when loaded', async () => {
    const component = mount(<App />, {
      wrappingComponent: Wrapper,
    });
    await act(async () => {
      await waitForExpect(() => {
        component.update();
        expect(component.text()).toEqual(
          expect.stringContaining('Hello, World!'));
      });
    });
    expect(component).toMatchSnapshot();
  });
});
