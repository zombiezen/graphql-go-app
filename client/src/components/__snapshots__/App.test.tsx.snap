// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a loading message 1`] = `
<Component>
  <main
    className="App"
  >
    <p>
      Loadingâ€¦
    </p>
  </main>
</Component>
`;

exports[`App renders the greeting when loaded 1`] = `
<Component>
  <main
    className="App"
  >
    <p>
      Hello, World!
    </p>
  </main>
</Component>
`;
